packages/ember-glimmer/tests/integration/syntax/with-test.js

```diff
@@ -30,13 +25,17 @@ moduleFor('Syntax test: {{#with as}}', class extends SharedSyntaxConditionalsTes
 
     this.assertText('Hello');
 
+    this.runTask(() => set(this.context, 'cond1.greeting', 'Hello world'));
+
+    this.assertText('Hello world');
+
     this.runTask(() => set(this.context, 'cond1', false));
 
     this.assertText('False');
 
-    this.runTask(() => this.rerender());
+    this.runTask(() => set(this.context, 'cond1', { greeting: 'Hello' }));
 
-    this.assertText('False');
+    this.assertText('Hello');
   }
```

Comment:
As per our usual protocol, we are supposed to be testing "update via mutation" and "reset via replacement" here.